soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -2612,8 +2625,39 @@ u8 Randomizer::GetRandoSettingValue(RandomizerSettingKey randoSettingKey) {
     return this->randoSettings[randoSettingKey];
 }
 
-GetItemID Randomizer::GetItemIdFromKnownCheck(RandomizerCheck randomizerCheck, GetItemID ogItemId) {
-    return GetItemIdFromRandomizerGet(this->itemLocations[randomizerCheck], ogItemId);
+GetItemEntry Randomizer::GetItemEntryFromRGData(RandomizerGetData rgData, GetItemID ogItemId, bool checkObtainability) {
+    // Go ahead and early return the ogItemId's entry if we somehow get RG_NONE.
+    if (rgData.rgID == RG_NONE) {
+        return ItemTableManager::Instance->RetrieveItemEntry(MOD_NONE, ogItemId);
+    }
+    if (checkObtainability && OTRGlobals::Instance->gRandomizer->GetItemObtainabilityFromRandomizerGet(rgData.rgID) != CAN_OBTAIN) {
+        return ItemTableManager::Instance->RetrieveItemEntry(MOD_NONE, GI_RUPEE_BLUE);
+    }
+    // Can't get RG_ICE_TRAP if the rgID corresponds to a vanilla item
+    if (IsItemVanilla(rgData.rgID)) {
+        return ItemTableManager::Instance->RetrieveItemEntry(MOD_NONE, GetItemIdFromRandomizerGet(rgData.rgID, ogItemId));
+    }
+    // After this point we can assume we are dealing with a randomizer exclusive item.
+    GetItemEntry giEntry = ItemTableManager::Instance->RetrieveItemEntry(MOD_RANDOMIZER, rgData.rgID);
+    // If we have an ice trap, we want to change the GID and DrawFunc to the fakeRgID's values.
+    if (rgData.rgID == RG_ICE_TRAP) {
```

Comment:
Not something we need to worry about yet, but comparing everything to ICE_TRAP is going to need to change at some point when we introduce even one more trap type

Reply:
True, that can be wrapped up into a function or something later.